### PR TITLE
GetDirectoryName fixed for path with no slash

### PR DESCRIPTION
**Pull request by dgGC5**
## Description
I have fixed the bug, when the -1 returned from method LastIndexOf() was a second parameter of method Substring, which obviosuly gave an error.

## Motivation and Context

I have checked the proposed change of behaviour with original C# and it seems to work as I propose.
Please check my snippet here: https://dotnetfiddle.net/ZuRy2E

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (work on Unit Tests, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

By the way:
Link https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md is not working for me (firefox).

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/nanoframework/System.IO.FileSystem?pullRequest=32) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/nanoframework/System.IO.FileSystem?pullRequest=32) before we can accept your contribution.<br/><br/>:x: dgGC5 [sign now](https://cla.dotnetfoundation.org/nanoframework/System.IO.FileSystem?pullRequest=32)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/nanoframework/System.IO.FileSystem?pullRequest=32) it.</sub>

File: System.IO.FileSystem/Path.cs [hunk 1]
```diff
@@ -139,6 +139,9 @@ public static string GetDirectoryName(string path)
 
                     var lastPathPostion = path.LastIndexOf(DirectorySeparatorChar);
 
+                    if (lastPathPostion == -1)
```
Comment (josesimoes) [inline System.IO.FileSystem/Path.cs:142]: Makes sense!

Please follow project code style and add open/close brackets to if clause.

Comment (dgGC5) [reply System.IO.FileSystem/Path.cs:142]: Sure, I have fixed my C# style about lacking of brackets.

However I read this document: https://docs.nanoframework.net/content/contributing/cs-coding-style.html and in the example file, perhaps in 18 line you do not use brackets for if clause. That is why I thought we can use short version of this syntax.

Anyway fixed :) Thank you for quick reply.

Comment (josesimoes) [reply System.IO.FileSystem/Path.cs:142]: You have a point... 
That example it's outdated. And we should have a config file for these rules anyway. 😞 

Thanks for the fix!

**Review by josesimoes — APPROVED** [review]
LGTM!

File: System.IO.FileSystem/Path.cs (hunk 1 above)
Comment (dgGC5) [reply System.IO.FileSystem/Path.cs:142]: +1 for config file rule
Thank you for cooperation.
Have a nice day :)

**Pull request merged**